content/blog/2022/2022-02-15-getting-started-with-spring-webflux.md

```diff
@@ -0,0 +1,566 @@
+                .andRoute(DELETE("/handler/users/{userId}").and(accept(MediaType.APPLICATION_JSON)), handler::deleteUserById);
+    }
+}
+```
+
+Finally, we will define some properties in order to configure our database connection and server config.
+
+```yaml
+spring:
+  application:
+    name: spring-webflux-guide
+  webflux:
+    base-path: /api
+  data:
+    mongodb:
+      authentication-database: admin
+      uri: mongodb://localhost:27017/test
+      database: test
+
+logging:
+  level:
+    io:
+      reflectoring: DEBUG
+    org:
+      springframework:
+        web: INFO
+        data:
+          mongodb:
+            core:
+              ReactiveMongoTemplate: DEBUG
+    reactor:
+      netty:
+        http:
+          client: DEBUG
+```
+
+This constitutes our basic non-blocking REST API using Spring Webflux. Now this works as a Publisher-Subscriber model that we were talking about initially in this article.
+
+### Server Sent events
+
+*Server Sent Events* is an HTTP standard which provides capability for servers to push streaming data to the web client. The flow is unidirectional from server to client and client receives updates whenever the server pushes some data. This kind of mechanism is often used for real-time messaging, streaming or notification of events or data. Usually, for multiplexed and bidirectional streaming, we often use *Websockets*. But SSE are mostly used for following use-cases:
+
+* Receiving live feed from the server whenever there is new or updated record.
+* Message notification without unnecessary reloading of server.
+* Subscribing to a feed of news, stocks or cryptocurrency
+
+The biggest limitation of SSE is that it’s unidirectional and hence information can’t be passed to a server from the client. Spring Wbflux allows us to define Server streaming events which can send events in a given interval. The web client initiates the REST API call and keeps it open until the even stream is closed. The server-side event would have the content type `text/event-stream`. Now we can define a Server Side Event streaming endpoint using WebFlux by simply returning a Flux and specifying the content type as text/event-stream. So let’s add this method to our existing `UserController`:
+
+```java
+    @GetMapping(value = "/stream", produces = MediaType.TEXT_EVENT_STREAM_VALUE)
+    public Flux<User> streamAllUsers() {
+        return userService
+                .getAllUsers()
+                .flatMap(user -> Flux
+                        .zip(Flux.interval(Duration.ofSeconds(2)),
+                                Flux.fromStream(Stream.generate(() -> user))
+                        )
+                        .map(Tuple2::getT2)
+                );
+    }
+```
+
+Here, we will stream all the users in our system every 2 seconds. This serves the whole list of updated users from the MongoDB every interval.
+
+### Brief introduction to WebClient
```

Comment:
This section looks a bit lost in the context of the article. Why are we talking about WebClient here when the article is about building a server? I'm leaning towards removing this section, especially since there is already an article about WebClient on reflectoring: https://reflectoring.io/spring-webclient/. Maybe just link to that article in the conclusion?

Reply:
Webclient is an integral part of Webflux architecture. So I thought we can add a brief introduction to Webclient. Maybe, we can keep this section short, reduce it and refer this(https://reflectoring.io/spring-webclient/) article for more information.